app/renderer/components/Inspector/HighlighterRects.js

```diff
@@ -1,20 +1,152 @@
 import React, { Component } from 'react';
 import HighlighterRect from './HighlighterRect';
+import HighlighterCentroid from './HighlighterCentroid';
 import B from 'bluebird';
-import { SCREENSHOT_INTERACTION_MODE } from './shared';
+import { SCREENSHOT_INTERACTION_MODE, parseCoordinates } from './shared';
 
 const {TAP, SWIPE, SELECT} = SCREENSHOT_INTERACTION_MODE;
+const RENDER = {CENTROID: 'centroid', EXPAND: 'expand', OVERLAP: 'overlap'};
 
 /**
  * Shows screenshot of running application and divs that highlight the elements' bounding boxes
  */
 export default class HighlighterRects extends Component {
 
+  getElements (source) {
+    const elementsByOverlap = this.findElementProps(source, null, [], {});
+    let elements = [];
+
+    for (let key in elementsByOverlap) {
+      let element = elementsByOverlap[key][0];
+
+      //Create new element obj which will be a +/- centroid
+      if (elementsByOverlap[key].length > 1) {
+        element = {
```

Comment:
is there a reason we're mutating `element` rather than creating a new object? we typically never want to mutate unless there's a reason